docs/hardhat-network/README.md

```diff
@@ -82,164 +85,37 @@ When a transaction fails without a reason, Hardhat Network will create a clear e
-
-- You can use it in calls and transactions. It works with `view` functions, but not in `pure` ones.
-- It always works, regardless of the call or transaction failing or being successful.
-- To use it you need to import `hardhat/console.sol`.
-- You can call `console.log` with up to 4 parameters in any order of following types:
-  - `uint`
-  - `string`
-  - `bool`
-  - `address`
-- There's also the single parameter API for the types above, and additionally `bytes`, `bytes1`... up to `bytes32`:
-  - `console.logInt(int i)`
-  - `console.logUint(uint i)`
-  - `console.logString(string memory s)`
-  - `console.logBool(bool b)`
-  - `console.logAddress(address a)`
-  - `console.logBytes(bytes memory b)`
-  - `console.logBytes1(bytes1 b)`
-  - `console.logBytes2(bytes2 b)`
-  - ...
-  - `console.logBytes32(bytes32 b)`
-- `console.log` implements the same formatting options that can be found in Node.js' [`console.log`](https://nodejs.org/dist/latest-v12.x/docs/api/console.html#console_console_log_data_args), which in turn uses [`util.format`](https://nodejs.org/dist/latest-v12.x/docs/api/util.html#util_util_format_format_args).
-  - Example: `console.log("Changing owner from %s to %s", currentOwner, newOwner)`
-- It works with any library: ethers.js, web3.js, waffle, truffle-contract, etc.
-- `console.log` is implemented in standard Solidity and then detected in Hardhat Network. This makes its compilation work with any other tools (like Remix, Waffle or Truffle).
-- `console.log` calls can run in other networks, like mainnet, kovan, ropsten, etc. They do nothing in those networks, but do spend a minimal amount of gas.
-
-## Mainnet forking
-
-The Hardhat Network is empty by default, except for some accounts with an initial balance. But sometimes it's more useful to have a local network that simulates the state of the mainnet. This is what forking is for.
-
-To fork from the mainnet you need the URL of a node to connect to. For example, using Alchemy, you can start a local node that forks the mainnet with this command:
+### `console.log`
 
-```
-npx hardhat node --fork https://eth-mainnet.alchemyapi.io/v2/<key>
-```
+Hardhat Network allows you to print logging messages and contract variables by calling `console.log()` from your Solidity code.
 
-where you have to replace `<key>` with your Alchemy API key.
+To use it, you simply import `hardhat/console.sol` and call it. It implements the same formatting options that can be found in Node.js' [`console.log`](https://nodejs.org/dist/latest-v12.x/docs/api/console.html#console_console_log_data_args), which in turn uses [`util.format`](https://nodejs.org/dist/latest-v12.x/docs/api/util.html#util_util_format_format_args). For example: `console.log("Changing owner from %s to %s", currentOwner, newOwner)`.
```

Comment:
I think we should explain here that this only works on hardhat (even if it can be inferred from the context), but that contracts will continue to work on mainnet (but that having console.log statements will use a bit of extra gas unnecessarily). This could be done in a notice/aside block.

Reply:
The next paragraph does explain that it only works on Hardhat (and Tenderly, per Pato), and that it's a no-op everywhere else.  I added a bit of extra text there about the gas: https://github.com/nomiclabs/hardhat/pull/1657/commits/170884778b0bb7eb261c8e4480c24296717eba23